tests/test_coadds.py

```diff
@@ -98,45 +100,45 @@ def setUpClass(cls) -> None:
         # and border_size = 5. If that is changed, these values need an update.
         sources = (
             # flux, centroid, shape
-            (1000.0, geom.Point2D(6.3, 7.2), None),
-            (2500.0, geom.Point2D(16.8, 18.3), None),
-            (1500.0, geom.Point2D(21.2, 5.1), None),
-            (3200.0, geom.Point2D(16.1, 23.9), None),
-            (1800.0, geom.Point2D(44.7, 8.9), None),
-            (2100.0, geom.Point2D(34.1, 19.2), None),
-            (900.0, geom.Point2D(9.1, 13.9), Quadrupole(2.5, 1.5, 0.8)),
-            (1250.0, geom.Point2D(19.3, 11.2), Quadrupole(1.5, 2.5, 0.75)),
-            (2100.0, geom.Point2D(5.1, 21.2), Quadrupole(1.7, 1.9, 0.05)),
-            (2800.0, geom.Point2D(24.1, 19.2), Quadrupole(1.9, 1.7, 0.1)),
-            (2350.0, geom.Point2D(40.3, 13.9), Quadrupole(1.8, 1.8, -0.4)),
-            (4999.0, geom.Point2D(45.8, 22.0), Quadrupole(1.6, 1.2, 0.2)),
+            (1000.0, geom.Point2D(cls.x0 + 6.3, cls.y0 + 7.2), None),
+            (2500.0, geom.Point2D(cls.x0 + 16.8, cls.y0 + 18.3), None),
+            (1500.0, geom.Point2D(cls.x0 + 21.2, cls.y0 + 5.1), None),
+            (3200.0, geom.Point2D(cls.x0 + 16.1, cls.y0 + 23.9), None),
+            (1800.0, geom.Point2D(cls.x0 + 44.7, cls.y0 + 8.9), None),
+            (2100.0, geom.Point2D(cls.x0 + 34.1, cls.y0 + 19.2), None),
+            (900.0, geom.Point2D(cls.x0 + 9.1, cls.y0 + 13.9), Quadrupole(2.5, 1.5, 0.8)),
+            (1250.0, geom.Point2D(cls.x0 + 19.3, cls.y0 + 11.2), Quadrupole(1.5, 2.5, 0.75)),
+            (2100.0, geom.Point2D(cls.x0 + 5.1, cls.y0 + 21.2), Quadrupole(1.7, 1.9, 0.05)),
+            (2800.0, geom.Point2D(cls.x0 + 24.1, cls.y0 + 19.2), Quadrupole(1.9, 1.7, 0.1)),
+            (2350.0, geom.Point2D(cls.x0 + 40.3, cls.y0 + 13.9), Quadrupole(1.8, 1.8, -0.4)),
+            (4999.0, geom.Point2D(cls.x0 + 45.8, cls.y0 + 22.0), Quadrupole(1.6, 1.2, 0.2)),
         )
 
         # The mapping of positions to cell indices assume inner_size = (17, 15)
         # and border_size = 5. If that is changed, these values need an update.
```

Comment:
How would they change (by a future developer changing this test?), and if it might change, wouldn't it be better to make these values variables?

Reply:
I made the mapping between locations and cell indices automated. If the `inner_size` or `border_size` are changed, the developer can pick any point that should satisfy the description in the in-line comment. The test would pass even if the developer doesn't change the points because of the automated mapping, but it would become weaker. The hardcoded mapping meant that the developer could not change values arbitrarily without knowing what they were doing, which made the test more robust IMO.